src/yunohost/utils/network.py

```diff
@@ -37,8 +39,8 @@ def get_public_ip(protocol=4):
         raise ValueError("invalid protocol version")
 
     try:
-        return urlopen(url).read().strip()
-    except IOError:
+        return requests.get(url, timeout=30).content.strip()
+    except ConnectionError:
```

Comment:
Uuuh shouldnt we cover other exception than ConnectionError ? We might want to use the helper `download_text` here : https://github.com/YunoHost/moulinette/blob/unstable/moulinette/utils/network.py#L8

Reply:
ah, that seems indeed way better

Reply:
Done